test/aries-js-worker/test/didexchange/edgeagent_router.js

```diff
@@ -135,23 +135,15 @@ describe("DID-Exchange between two Edge Agents using the router", function () {
     var invitation, connectionID
 
     before(async () => {
-        return new Promise((resolve, reject) => {
```

Comment:
what was the issue?

Reply:
https://github.com/hyperledger/aries-framework-go/pull/1387/checks?check_run_id=474809170#step:5:501

Reply:
Perhaps more investigation is needed (learning curve).

This looks like a simple refactor to my eyes.

Mocha also waits until the promise returned by an async function passed to `before()` to resolve.

The build passed on PR #1392

Reply:
Here is the merge build from PR #1392: https://github.com/hyperledger/aries-framework-go/actions/runs/46404517. Tests passed.

Reply:
I notice that [your branch](https://github.com/soluchok/aries-framework-go/commits/issue-1173) was lagging behind a bit when it got merged today.

Reply:
I can't figure out what exactly was the issue.

Reply:
So, the problem here is that we have created the promise and then we are resolving it twice. It means after first resolve it is done. It is not waiting for the second one. It does not give us a guarantee that two agents were started successfully. I just not lucky enough :)

```
 return new Promise((resolve, reject) => {
      newAries('alice').then( ..  resolve() ..)
      newAries('alice').then( ..  resolve() ..)
}
```

Reply:
Ah! I got it. The promise was being resolved when the `alice` agent was fetch but *before* bob's agent was loaded.

Reply:
Yeah, but I am not :)

Reply:
@soluchok can you push a separate PR for this fix?

Reply:
Done!